main.tf

```diff
@@ -23,12 +23,14 @@ resource "azurerm_network_security_rule" "predefined_rules" {
   source_port_ranges                         = split(",", replace(lookup(var.predefined_rules[count.index], "source_port_range", "*"), "*", "0-65535"))
   destination_port_range                     = element(var.rules[lookup(var.predefined_rules[count.index], "name")], 4)
   description                                = element(var.rules[lookup(var.predefined_rules[count.index], "name")], 5)
-  source_address_prefix                      = length(lookup(var.predefined_rules[count.index], "source_application_security_group_ids", [])) == 0 ? join(",", var.source_address_prefix) : ""
-  destination_address_prefix                 = length(lookup(var.predefined_rules[count.index], "destination_application_security_group_ids", [])) == 0 ? join(",", var.destination_address_prefix) : ""
+  source_address_prefix                      = length(lookup(var.predefined_rules[count.index], "source_application_security_group_ids", [])) == 0 && length(lookup(var.predefined_rules[count.index], "source_address_prefixes", [])) == 0 ? lookup(var.predefined_rules[count.index], "source_address_prefix", "*") : null
+  destination_address_prefix                 = length(lookup(var.predefined_rules[count.index], "destination_application_security_group_ids", [])) == 0 && length(lookup(var.predefined_rules[count.index], "destination_address_prefixes", [])) == 0 ? lookup(var.predefined_rules[count.index], "destination_address_prefix", "*") : null
+  source_address_prefixes                    = length(lookup(var.predefined_rules[count.index], "source_application_security_group_ids", [])) == 0 ? lookup(var.predefined_rules[count.index], "source_address_prefixes", null) : null
+  destination_address_prefixes               = length(lookup(var.predefined_rules[count.index], "destination_application_security_group_ids", [])) == 0 ? lookup(var.predefined_rules[count.index], "destination_address_prefixes", null) : null
```

Comment:
1. `predefined_rules` uses `var.source_address_prefix` instead of `source_address_prefix` in `predefined_rules`, would you mind using `var.source_address_prefix` in order not to break users?
2. Would you mind only using `source_address_prefixes` in `azurerm_network_security_rule` instead of the two to simplify the code?

Reply:
1. Using a specific `source_address_prefix` and `destination_address_prefix` (as well as the plural versions) makes defining deviating rules possible.
2. Yes, I would mind, since that restricts the possibility of being as specific as possible